I got a panic error when executing `okteto up` and my `resources.requests` are too high

**Describe the bug**

I had deployed the [`go-getting-started`](https://github.com/okteto/go-getting-started) example and modified the okteto manifest to set some requests to me dev container and when I executed `okteto up` I'm getting a panic error:

```
okteto up
 i  Using ifbyol @ xxxxxxxxx as context
 i  'go-getting-started' was already deployed. To redeploy run 'okteto deploy' or 'okteto up --deploy'
 i  Images were already built. To rebuild your images run 'okteto build' or 'okteto deploy --build'
panic: runtime error: index out of range [1] with length 0

goroutine 87 [running]:
github.com/okteto/okteto/pkg/k8s/deployments.getResourceLimitError({0xc0001d2a80, 0x75}, 0xc001d52600)
	github.com/okteto/okteto/pkg/k8s/deployments/crud.go:186 +0x3f3
github.com/okteto/okteto/pkg/k8s/deployments.CheckConditionErrors(0x3c67f30?, 0xc001c74480?)
	github.com/okteto/okteto/pkg/k8s/deployments/crud.go:158 +0x331
github.com/okteto/okteto/pkg/k8s/apps.(*DeploymentApp).CheckConditionErrors(0xc001c64198?, 0x3c43bb0?)
	github.com/okteto/okteto/pkg/k8s/apps/deployments.go:111 +0x1d
github.com/okteto/okteto/pkg/k8s/apps.GetRunningPodInLoop({0x3c43bb0, 0xc0000b0690}, 0xc001d52600, {0x3c56448, 0xc001c64198}, {0x3c67f30, 0xc001c74480})
	github.com/okteto/okteto/pkg/k8s/apps/crud.go:73 +0x107
github.com/okteto/okteto/cmd/up.(*upContext).createDevContainer(0xc0000de160, {0x3c43bb0, 0xc0000b0690}, {0x3c56448, 0xc000193e00}, 0x0)
	github.com/okteto/okteto/cmd/up/activate.go:308 +0x76d
github.com/okteto/okteto/cmd/up.(*upContext).devMode(0xc0000de160?, {0x3c43bb0, 0xc0000b0690}, {0x3c56448, 0xc000193e00}, 0x6?)
	github.com/okteto/okteto/cmd/up/activate.go:246 +0x32
github.com/okteto/okteto/cmd/up.(*upContext).activate(0xc0000de160)
	github.com/okteto/okteto/cmd/up/activate.go:138 +0xaa5
github.com/okteto/okteto/cmd/up.(*upContext).activateLoop(0xc0000de160)
	github.com/okteto/okteto/cmd/up/up.go:705 +0x296
created by github.com/okteto/okteto/cmd/up.(*upContext).start
	github.com/okteto/okteto/cmd/up/up.go:637 +0x6e5
```

Checking the stack trace, it seems we are trying to access to an element of an empty array. It seems the error is because the memory I specified is too high. I wasn't able to reproduce it with other examples like movies.

**To Reproduce**
Steps to reproduce the behavior:

1. Deploy go-getting-started
2. Modify the okteto manifest to specify the following: `resources.requests.cpu`=`200m` and `resources.requests.memory`=`10Gi`
3. Execute `okteto up`

**Expected behavior**

I shouldn't get a panic. If the operation cannot be done for whatever reason I should get a proper error message

**Desktop (please complete the following information):**

- Version: `2.18.0`

**Additional context**
N/A

--- a/pkg/k8s/deployments/crud.go
+++ b/pkg/k8s/deployments/crud.go
@@ -172,10 +172,11 @@ func isResourcesRelatedError(errorMessage string) bool {
 
 func getResourceLimitError(errorMessage string, dev *model.Dev) error {
 	var errorToReturn string
+	var regexMaxSubstring = 2
 	if strings.Contains(errorMessage, "maximum cpu usage") {
 		cpuMaximumRegex := regexp.MustCompile(`cpu usage per Pod is (\d*\w*)`)
 		maximumCpuPerPodMatchGroups := cpuMaximumRegex.FindStringSubmatch(errorMessage)
-		if len(maximumCpuPerPodMatchGroups) < 2 {
+		if len(maximumCpuPerPodMatchGroups) < regexMaxSubstring {
 			errorToReturn += "The value of resources.limits.cpu in your okteto manifest exceeds the maximum CPU limit per pod. "
 		} else {
 			var manifestCpu string
@@ -189,7 +190,7 @@ func getResourceLimitError(errorMessage string, dev *model.Dev) error {
 	if strings.Contains(errorMessage, "maximum memory usage") {
 		memoryMaximumRegex := regexp.MustCompile(`memory usage per Pod is (\d*\w*)`)
 		maximumMemoryPerPodMatchGroups := memoryMaximumRegex.FindStringSubmatch(errorMessage)
-		if len(maximumMemoryPerPodMatchGroups) < 2 {
+		if len(maximumMemoryPerPodMatchGroups) < regexMaxSubstring {
 			errorToReturn += "The value of resources.limits.memory in your okteto manifest exceeds the maximum memory limit per pod."
 		} else {
 			var manifestMemory string

--- a/pkg/k8s/deployments/crud.go
+++ b/pkg/k8s/deployments/crud.go
@@ -176,7 +176,7 @@ func getResourceLimitError(errorMessage string, dev *model.Dev) error {
 		cpuMaximumRegex := regexp.MustCompile(`cpu usage per Pod is (\d*\w*)`)
 		maximumCpuPerPodMatchGroups := cpuMaximumRegex.FindStringSubmatch(errorMessage)
 		if len(maximumCpuPerPodMatchGroups) < 2 {
-			errorToReturn += fmt.Sprintf("The value of resources.limits.cpu in your okteto manifest exceeds the maximum CPU limit per pod. ")
+			errorToReturn += "The value of resources.limits.cpu in your okteto manifest exceeds the maximum CPU limit per pod. "
 		} else {
 			var manifestCpu string
 			if limitCpu, ok := dev.Resources.Limits[apiv1.ResourceCPU]; ok {
@@ -190,7 +190,7 @@ func getResourceLimitError(errorMessage string, dev *model.Dev) error {
 		memoryMaximumRegex := regexp.MustCompile(`memory usage per Pod is (\d*\w*)`)
 		maximumMemoryPerPodMatchGroups := memoryMaximumRegex.FindStringSubmatch(errorMessage)
 		if len(maximumMemoryPerPodMatchGroups) < 2 {
-			errorToReturn += fmt.Sprintf("The value of resources.limits.memory in your okteto manifest exceeds the maximum memory limit per pod.")
+			errorToReturn += "The value of resources.limits.memory in your okteto manifest exceeds the maximum memory limit per pod."
 		} else {
 			var manifestMemory string
 			if limitMemory, ok := dev.Resources.Limits[apiv1.ResourceMemory]; ok {

--- a/pkg/k8s/deployments/crud.go
+++ b/pkg/k8s/deployments/crud.go
@@ -174,21 +174,31 @@ func getResourceLimitError(errorMessage string, dev *model.Dev) error {
 	var errorToReturn string
 	if strings.Contains(errorMessage, "maximum cpu usage") {
 		cpuMaximumRegex := regexp.MustCompile(`cpu usage per Pod is (\d*\w*)`)
-		maximumCpuPerPod := cpuMaximumRegex.FindStringSubmatch(errorMessage)[1]
-		var manifestCpu string
-		if limitCpu, ok := dev.Resources.Limits[apiv1.ResourceCPU]; ok {
-			manifestCpu = limitCpu.String()
+		maximumCpuPerPodMatchGroups := cpuMaximumRegex.FindStringSubmatch(errorMessage)
+		if len(maximumCpuPerPodMatchGroups) < 2 {
+			errorToReturn += fmt.Sprintf("The value of resources.limits.cpu in your okteto manifest exceeds the maximum CPU limit per pod. ")
+		} else {
+			var manifestCpu string
+			if limitCpu, ok := dev.Resources.Limits[apiv1.ResourceCPU]; ok {
+				manifestCpu = limitCpu.String()
+			}
+			maximumCpuPerPod := maximumCpuPerPodMatchGroups[1]
+			errorToReturn += fmt.Sprintf("The value of resources.limits.cpu in your okteto manifest (%s) exceeds the maximum CPU limit per pod (%s). ", manifestCpu, maximumCpuPerPod)
 		}
-		errorToReturn += fmt.Sprintf("The value of resources.limits.cpu in your okteto manifest (%s) exceeds the maximum CPU limit per pod (%s). ", manifestCpu, maximumCpuPerPod)
 	}
 	if strings.Contains(errorMessage, "maximum memory usage") {
 		memoryMaximumRegex := regexp.MustCompile(`memory usage per Pod is (\d*\w*)`)
-		maximumMemoryPerPod := memoryMaximumRegex.FindStringSubmatch(errorMessage)[1]
-		var manifestMemory string
-		if limitMemory, ok := dev.Resources.Limits[apiv1.ResourceMemory]; ok {
-			manifestMemory = limitMemory.String()
+		maximumMemoryPerPodMatchGroups := memoryMaximumRegex.FindStringSubmatch(errorMessage)
+		if len(maximumMemoryPerPodMatchGroups) < 2 {
+			errorToReturn += fmt.Sprintf("The value of resources.limits.memory in your okteto manifest exceeds the maximum memory limit per pod.")
+		} else {
+			var manifestMemory string
+			if limitMemory, ok := dev.Resources.Limits[apiv1.ResourceMemory]; ok {
+				manifestMemory = limitMemory.String()
+			}
+			maximumMemoryPerPod := maximumMemoryPerPodMatchGroups[1]
+			errorToReturn += fmt.Sprintf("The value of resources.limits.memory in your okteto manifest (%s) exceeds the maximum memory limit per pod (%s). ", manifestMemory, maximumMemoryPerPod)
 		}
-		errorToReturn += fmt.Sprintf("The value of resources.limits.memory in your okteto manifest (%s) exceeds the maximum memory limit per pod (%s). ", manifestMemory, maximumMemoryPerPod)
 	}
 	return fmt.Errorf(strings.TrimSpace(errorToReturn))
 }

--- a/pkg/k8s/deployments/crud_test.go
+++ b/pkg/k8s/deployments/crud_test.go
@@ -474,7 +474,7 @@ func TestCheckConditionErrors(t *testing.T) {
 				t.Fatalf("wrong error. Got nil, expected %s", tt.expectedErr.Error())
 			}
 			if err != nil && tt.expectedErr == nil {
-				t.Fatalf("wrong error. Got nil, expected %s", err.Error())
+				t.Fatalf("wrong error. Expected nil, but got %s", err.Error())
 			}
 		})
 	}

--- a/pkg/k8s/deployments/crud_test.go
+++ b/pkg/k8s/deployments/crud_test.go
@@ -279,6 +279,62 @@ func TestCheckConditionErrors(t *testing.T) {
 			fmt.Errorf("The value of resources.limits.memory in your okteto manifest (5Gi) exceeds the maximum memory limit per pod (3Gi)."),
 		},
 		{
+			"Memory per pod exceeded 2",
+			&appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "fake",
+					Namespace: "test",
+				},
+				Status: appsv1.DeploymentStatus{
+					Conditions: []appsv1.DeploymentCondition{
+						{
+							Type:    appsv1.DeploymentReplicaFailure,
+							Reason:  "FailedCreate",
+							Status:  apiv1.ConditionTrue,
+							Message: "maximum memory usage",
+						},
+					},
+				},
+			},
+			&model.Dev{
+				Resources: model.ResourceRequirements{
+					Limits: model.ResourceList{
+						apiv1.ResourceCPU:    resource.MustParse("2"),
+						apiv1.ResourceMemory: resource.MustParse("5Gi"),
+					},
+				},
+			},
+			fmt.Errorf("The value of resources.limits.memory in your okteto manifest (5Gi) exceeds the maximum memory limit per pod (3Gi)."),
+		},
+		{
+			"Cpu per pod exceeded",
+			&appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "fake",
+					Namespace: "test",
+				},
+				Status: appsv1.DeploymentStatus{
+					Conditions: []appsv1.DeploymentCondition{
+						{
+							Type:    appsv1.DeploymentReplicaFailure,
+							Reason:  "FailedCreate",
+							Status:  apiv1.ConditionTrue,
+							Message: "maximum cpu usage",
+						},
+					},
+				},
+			},
+			&model.Dev{
+				Resources: model.ResourceRequirements{
+					Limits: model.ResourceList{
+						apiv1.ResourceCPU:    resource.MustParse("2"),
+						apiv1.ResourceMemory: resource.MustParse("5Gi"),
+					},
+				},
+			},
+			fmt.Errorf("The value of resources.limits.cpu in your okteto manifest (2) exceeds the maximum CPU limit per pod (1)."),
+		},
+		{
 			"Cpu per pod exceeded",
 			&appsv1.Deployment{
 				ObjectMeta: metav1.ObjectMeta{
@@ -334,19 +390,91 @@ func TestCheckConditionErrors(t *testing.T) {
 			},
 			fmt.Errorf("The value of resources.limits.cpu in your okteto manifest (2) exceeds the maximum CPU limit per pod (1). The value of resources.limits.memory in your okteto manifest (5Gi) exceeds the maximum memory limit per pod (3Gi)."),
 		},
+		{
+			name: "exceeded storage quota",
+			deployment: &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "fake",
+					Namespace: "test",
+				},
+				Status: appsv1.DeploymentStatus{
+					Conditions: []appsv1.DeploymentCondition{
+						{
+							Type:    appsv1.DeploymentReplicaFailure,
+							Reason:  "FailedCreate",
+							Status:  apiv1.ConditionTrue,
+							Message: "exceeded quota: requested: requests.storage=1Gi, used: requests.storage=2Gi, limited: requests.storage=3Gi",
+						},
+					},
+				},
+			},
+			dev:         &model.Dev{},
+			expectedErr: fmt.Errorf("quota exceeded, you have reached the maximum storage per namespace"),
+		},
+		{
+			name: "exceeded number of pods quota",
+			deployment: &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "fake",
+					Namespace: "test",
+				},
+				Status: appsv1.DeploymentStatus{
+					Conditions: []appsv1.DeploymentCondition{
+						{
+							Type:    appsv1.DeploymentReplicaFailure,
+							Reason:  "FailedCreate",
+							Status:  apiv1.ConditionTrue,
+							Message: "exceeded quota: requested: pods=1, used: pods=2, limited: pods=3",
+						},
+					},
+				},
+			},
+			dev:         &model.Dev{},
+			expectedErr: fmt.Errorf("quota exceeded, you have reached the maximum number of pods per namespace"),
+		},
+		{
+			name: "another error",
+			deployment: &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "fake",
+					Namespace: "test",
+				},
+				Status: appsv1.DeploymentStatus{
+					Conditions: []appsv1.DeploymentCondition{
+						{
+							Type:    appsv1.DeploymentReplicaFailure,
+							Reason:  "FailedCreate",
+							Status:  apiv1.ConditionTrue,
+							Message: "another error",
+						},
+					},
+				},
+			},
+			dev:         &model.Dev{},
+			expectedErr: fmt.Errorf("another error"),
+		},
+		{
+			name: "No errors",
+			deployment: &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "fake",
+					Namespace: "test",
+				},
+			},
+			dev:         &model.Dev{},
+			expectedErr: nil,
+		},
 	}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-
 			err := CheckConditionErrors(tt.deployment, tt.dev)
 
-			if err == nil {
-				t.Fatalf("Didn't receive any error. Expected %s", tt.expectedErr)
+			if err == nil && tt.expectedErr != nil {
+				t.Fatalf("wrong error. Got nil, expected %s", tt.expectedErr.Error())
 			}
-
-			if err.Error() != tt.expectedErr.Error() {
-				t.Fatalf("wrong error. Got %s, expected %s", err, tt.expectedErr)
+			if err != nil && tt.expectedErr == nil {
+				t.Fatalf("wrong error. Got nil, expected %s", err.Error())
 			}
 		})
 	}

